model/riscv_sys_regs.sail

```diff
@@ -567,6 +567,38 @@ function retire_instruction() -> unit = {
        then minstret = minstret + 1
 }
 
+/*
+ * NOTE: This would be better placed in riscv_platform.sail, but that file
+ *       appears _after_ this one in the compile order meaning the valspec
+ *       for this function is unavailable when it's first encountered here.
+ *       Hence it appears here.
+ */
+val plat_enable_svadu = {ocaml: "Platform.enable_svadu",
+                         interpreter: "Platform.enable_svadu",
+                         c: "plat_enable_svadu",
+                         lem: "plat_enable_svadu"} : unit -> bool
+
+bitfield MEnvCfg : bits(64) = {
+  HADE : 61
+  /* Fill in other fields when adding support for them */
+}
+register menvcfg : MEnvCfg
+
+function legalize_menvcfg(c : MEnvCfg, high : bool, value : xlenbits) -> MEnvCfg = {
+  if not(high) & sizeof(xlen) == 32 then
+    c
+  else {
+    /* If 32bit, then high is true */
+    let full : bits(64) = match sizeof(xlen) {
+      64 => value,
+      32 => value @ 0x0000_0000
+    };
+    if not(plat_enable_svadu())
```

Comment:
Isn't this inverted? We should be able to toggle HADE if svadu is enabled.

Reply:
Good catch. I inverted this and cleaned up the `legalize_menvcfg` function. Thank you!